src/experimental/tools/download2go/index.ts

```diff
@@ -0,0 +1,415 @@
+/**
+ * Copyright 2019 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IDBPDatabase } from "idb";
+import { AsyncSubject } from "rxjs";
+
+import EventEmitter from "../../../utils/event_emitter";
+import PPromise from "../../../utils/promise";
+
+import { IActiveDownload, IActivePauses } from "./api/context/types";
+import { setUpDb } from "./api/db/dbSetUp";
+import DownloadManager from "./api/downloader/downloadManager";
+import {
+  getBuilderFormattedForAdaptations,
+  getBuilderFormattedForSegments,
+  getKeySystemsSessionsOffline,
+  offlineManifestLoader,
+} from "./api/downloader/manifest";
+import { IContentProtection } from "./api/drm/types";
+import {
+  IContentLoader,
+  IDownload2GoEvents,
+  IEmitterTrigger,
+  IGlobalSettings,
+  IInitSettings,
+  IStoredManifest,
+} from "./types";
+import {
+  checkForPauseAMovie,
+  checkForResumeAPausedMovie,
+  checkInitDownloaderOptions,
+  IndexDBError,
+  isSupported,
+  SegmentConstuctionError,
+  ValidationArgsError,
+} from "./utils";
+
+/**
+ * Instanciate a D2G downloader.
+ * @param {Object<{nameDB, storeManifestEvery}>} IOptionsStarter
+ * @return {IPublicAPI} IPublicAPI
+ */
+class D2G extends EventEmitter<IDownload2GoEvents> {
```

Comment:
Why not `Download2Go` or something more generic like `ContentDownloader`?

Reply:
Totally right! I haven't thought about this before but now that u say it, it totally make sense.